controller/volume_controller.go

```diff
@@ -623,6 +631,11 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 			// 3) The volume is being expanding.
 			if !(v.Status.InitialRestorationRequired || v.Status.IsStandby || v.Status.ExpansionRequired) {
 				v.Status.CurrentNodeID = ""
+				restoreCondition := types.GetCondition(v.Status.Conditions, types.VolumeConditionTypeRestore)
+				if restoreCondition.Reason == types.VolumeConditionReasonRestoreInProgress && restoreCondition.Status == types.ConditionStatusTrue {
```

Comment:
Check the Status first then the Reason. It doesn't make sense to check the reason if Status is not True.

Same for the below and following commits.